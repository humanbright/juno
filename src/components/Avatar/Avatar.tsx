import clsx from 'clsx';
import React, { ComponentType, forwardRef, useMemo } from 'react';

import {
  CustomStyledComponentResult,
  RcBaseProps,
  RcBaseSize,
  RcClassesProps,
  RcPaletteProp,
  styled,
  useThemeProps,
  withDeprecatedCheck,
} from '../../foundation';
import { SvgSymbol } from '../Icon';
import { RcPresence, RcPresenceProps } from '../Presence';
import { RcTooltip, withTooltip, WithTooltipProps } from '../Tooltip';
import {
  RcAvatarMask,
  StyledAvatar,
  StyledAvatarWrapper,
  StyledIconAvatar,
  StyledPresenceWrapper,
} from './styles';
import { RcAvatarClasses, RcAvatarSizes } from './utils';

type RcAvatarSize = RcBaseSize<
  'xxsmall' | 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge'
>;

type RcAvatarClassKey = 'presenceWrapper' | 'avatarContainer' | 'mask';

type PresenceOrigin = {
  vertical: 'top' | 'bottom';
  horizontal: 'left' | 'right';
};

type RcAvatarProps<T extends boolean = true> = {
  /** The anchor of the presence. */
  presenceOrigin?: PresenceOrigin;
  /** src for avatar image */
  src?: string;
  /** attrs on img element */
  imgProps?: React.ImgHTMLAttributes<HTMLImageElement>;
  /** avatar size, also presence size if not custom presence */
  size?: RcAvatarSize;
  /** avatar color */
  color?: RcPaletteProp;
  /** is that avatar can be click, default is false */
  clickable?: T;
  /**
   * with mack in the bottom of avatar,
   * when pass boolean, that will use default `<RcIcon symbol={Edit} />` to render mask,
   * also can provide `jsx` to custom below render
   */
  mask?: boolean | JSX.Element;
  /** support custom presence */
  presence?: JSX.Element;
  /** useless when prop 'presence' is not empty */
  presenceProps?: Omit<RcPresenceProps, 'size' | 'borderSize'>;
  /** icon for `RcIcon` render */
  iconSymbol?: SvgSymbol;
  /** can custom the icon size, half mean that will be half size of avatar */
  iconSize?: RcAvatarSize | 'half';
  /** only show presence when hovered or active */
  shouldRenderPresenceHovered?: boolean;

  /**
   * @deprecated
   * support custom Img component, if render img in avatar content
   * useless when 'iconSymbol' is provide,
   * just use `<img src /> in children`
   */
  Img?: ComponentType<{ src: string }>;
  /** @deprecated if 'true', will just use color value, use `color` directly */
  customColor?: boolean;
  /** @deprecated force set avatar clickable, default is true, please use clickable to set clickable */
  unClickable?: boolean;
  /** @deprecated not used, just keep type temp */
  isMember?: boolean;
  /** @deprecated not used, just keep type temp */
  displayName?: string;
  /** @deprecated alt for img, just use `<img src /> in children or imgProps for apply alt` */
  alt?: string;
  /** @deprecated support tooltip for avatar, use title to replace */
  tooltip?: string;
} & RcClassesProps<RcAvatarClassKey> &
  RcBaseProps<
    T extends true
      ? React.DetailedHTMLProps<
          React.ButtonHTMLAttributes<HTMLButtonElement>,
          HTMLButtonElement
        >
      : React.DetailedHTMLProps<
          React.HTMLAttributes<HTMLDivElement>,
          HTMLDivElement
        >,
    'color' | 'title'
  >;

const _RcAvatar = forwardRef<any, RcAvatarProps<true>>(
  (inProps: RcAvatarProps, ref) => {
    // change default unClickable to here, prevent warning
    const props = useThemeProps({ props: inProps, name: 'RcAvatar' });

    const {
      presenceOrigin,
      id: idProps,
      size,
      classes,
      color,
      children,
      clickable: _clickable,
      onClick,
      mask,
      presence,
      customColor,
      tooltip,
      iconSize,
      iconSymbol,
      src: srcProp,
      alt: altProp,
      Img,
      imgProps,
      presenceProps,
      displayName,
      ...rest
    } = props;

    let unClickable = props.unClickable;

    unClickable = unClickable ?? true;

    // TODO: only when use not set clickable, use the unClickable as click props,
    // TODO: let should remove when app replace all of that
    const clickable = _clickable ?? !unClickable;

    const handleClick = useMemo(() => {
      const haveClick = clickable || (!unClickable && onClick);

      return haveClick ? onClick : undefined;
    }, [clickable, onClick, unClickable]);

    const avatar = useMemo(() => {
      const avatarChildren = (() => {
        const src = srcProp || imgProps?.src;

        if (src) {
          const ImgComp = Img || 'img';
          return <ImgComp src={src} alt={altProp} {...imgProps} />;
        }

        if (iconSymbol) {
          const _size =
            iconSize === 'half'
              ? RcAvatarSizes[size!] / 2
              : RcAvatarSizes[iconSize || size!];

          return <StyledIconAvatar size={_size} symbol={iconSymbol} />;
        }

        return children || null;
      })();

      let _avatar = (
        <StyledAvatar
          size={size}
          color={color}
          iconSymbol={iconSymbol}
          className={clsx(
            classes?.avatarContainer,
            RcAvatarClasses.avatarContainer,
          )}
        >
          {avatarChildren}
          {mask && (
            <RcAvatarMask className={clsx(classes?.mask, RcAvatarClasses.mask)}>
              {typeof mask === 'boolean' ? undefined : mask}
            </RcAvatarMask>
          )}
        </StyledAvatar>
      );

      // TODO: this tooltip will be remove in Juno
      if (tooltip) {
        _avatar = <RcTooltip title={tooltip}>{_avatar}</RcTooltip>;
      }

      return _avatar;
    }, [
      Img,
      altProp,
      children,
      classes?.avatarContainer,
      classes?.mask,
      color,
      iconSize,
      iconSymbol,
      imgProps,
      mask,
      size,
      srcProp,
      tooltip,
    ]);

    const { horizontal, vertical } = presenceOrigin!;

    const presenceChildren = useMemo(() => {
      let _presence: JSX.Element | null = null;
      if (presence) {
        _presence = presence;
      } else if (presenceProps) {
        _presence = <RcPresence size={size} {...presenceProps} />;
      }

      if (_presence) {
        return (
          <StyledPresenceWrapper
            horizontal={horizontal}
            vertical={vertical}
            className={clsx(
              classes?.presenceWrapper,
              RcAvatarClasses.presenceWrapper,
            )}
          >
            {_presence}
          </StyledPresenceWrapper>
        );
      }

      return null;
    }, [classes, horizontal, presence, presenceProps, size, vertical]);

    /**
     * @deprecated
     * TODO: should remove in Juno
     * */
    const dataAttrProps = idProps
      ? {
          id: `${idProps}`,
          'data-uid': `${idProps}`,
          'data-cid': `${idProps}`,
          'data-test-automation-value': `${idProps}`,
        }
      : {};

    return (
      <StyledAvatarWrapper
        // as any for generic type
        ref={ref as any}
        {...dataAttrProps}
        onClick={handleClick}
        data-test-automation-class="avatar"
        // as any for generic type
        clickable={clickable as any}
        {...rest}
      >
        {avatar}
        {presenceChildren}
      </StyledAvatarWrapper>
    );
  },
);

const RcAvatar = styled(
  withDeprecatedCheck(
    withTooltip(_RcAvatar),
    [
      {
        prop: 'customColor',
        time: '2021-3',
        comment: `if 'true', will just use color value, use \`color\` directly `,
      },
      {
        prop: 'unClickable',
        time: '2021-3',
        comment: `force set avatar clickable, default is true, please use clickable to set clickable `,
      },
      {
        prop: 'isMember',
        time: '2021-3',
        comment: `not used, just keep type temp `,
      },
      {
        prop: 'displayName',
        time: '2021-3',
        comment: `not used, just keep type temp `,
      },
      {
        prop: 'alt',
        time: '2021-3',
        comment: `alt for img, just use \`<img src /> in children or imgProps for apply alt\` `,
      },
      {
        prop: 'tooltip',
        time: '2021-3',
        comment: `support tooltip for avatar, use title to replace `,
      },
    ],
    'RcAvatar',
  ),
)``;

RcAvatar.defaultProps = {
  size: 'medium',
  presenceOrigin: {
    horizontal: 'right',
    vertical: 'bottom',
  },
};

RcAvatar.displayName = 'RcAvatar';

const ExportType: <T extends boolean>(
  props: RcAvatarProps<T> & WithTooltipProps,
) => JSX.Element &
  CustomStyledComponentResult<
    RcAvatarProps<T> & WithTooltipProps
  > = RcAvatar as any;

export {
  ExportType as RcAvatar,
  RcAvatarProps,
  RcAvatarSize,
  StyledAvatar,
  PresenceOrigin,
};
