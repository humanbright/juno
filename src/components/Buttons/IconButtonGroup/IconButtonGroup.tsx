import React, { forwardRef, KeyboardEvent, MouseEvent } from 'react';

import {
  RcBaseDirection,
  RcTheme,
  styled,
  useA11yKeyEvent,
  useEventCallback,
  useThemeProps,
} from '../../../foundation';
import { IconButtonGroupStyle } from './styles';

type RcIconButtonGroupProps = {
  /** radius token for `border-radius` */
  radius?: keyof RcTheme['radius'];
  /** Define the padding size of that group wrapper. */
  space?: number | number[];
  /** Define the size of the gap between below each components. */
  gap?: number;
  /** The arrangement of components, default is `horizontal`  */
  direction?: RcBaseDirection;
  /** is Stop Propagation for the bar click */
  isStopPropagation?: boolean;
  /** component for render root container, default is `div` */
  component?: React.ElementType;
} & React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

const _RcIconButtonGroup = forwardRef<any, RcIconButtonGroupProps>(
  (inProps, ref) => {
    const props = useThemeProps({ props: inProps, name: 'RcIconButtonGroup' });
    const {
      children,
      isStopPropagation,
      onClick,
      onKeyUp,
      gap,
      space,
      radius,
      component: Component = 'div',
      ...rest
    } = props;
    const clickHandler = useEventCallback(
      (event: MouseEvent<HTMLDivElement>) => {
        onClick?.(event);
        isStopPropagation && event.stopPropagation();
      },
    );

    const handleOnKeyUp = useA11yKeyEvent(clickHandler, {
      preventDefault: false,
    });

    /** that button bar just for put buttons inside, and can stop below all events from this bar layer. */
    const keyUpHandler = useEventCallback(
      (event: KeyboardEvent<HTMLDivElement>) => {
        onKeyUp?.(event);
        isStopPropagation && event.stopPropagation();
        handleOnKeyUp(event);
      },
    );

    return (
      <Component
        onClick={clickHandler}
        onKeyUp={keyUpHandler}
        ref={ref}
        {...rest}
      >
        {children}
      </Component>
    );
  },
);

/**
 * provide IconButton group with wrapper
 */
const RcIconButtonGroup = styled(_RcIconButtonGroup)`
  ${IconButtonGroupStyle}
`;

RcIconButtonGroup.defaultProps = {
  gap: 0,
  direction: 'horizontal',
};

RcIconButtonGroup.displayName = 'RcIconButtonGroup';

export { RcIconButtonGroup, RcIconButtonGroupProps };
