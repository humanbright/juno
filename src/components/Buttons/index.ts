export * from './Button';
export * from './ButtonBar';
export * from './ButtonBase';
export * from './ButtonGroup';
export * from './FabButton';
export * from './FabIconButton';
export * from './IconButton';
export * from './IconButtonGroup';
export * from './RoundButton';
export * from './SplitButton';
export * from './ToggleButton';
export * from './ToggleButtonGroup';
