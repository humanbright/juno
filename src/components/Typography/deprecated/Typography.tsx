import MuiTypography, {
  TypographyProps as MuiTypographyProps,
} from '@material-ui/core/Typography';
import React, { ElementType } from 'react';

import { typography } from '../../../foundation';
import styled from '../../../foundation/styled-components';
import { RcBaseProps } from '../../../foundation/typings';

export type RcTypographyVariant = MuiTypographyProps['variant'];

type RcTypographyProps = {
  dependencies?: any[];
  variant?: RcTypographyVariant;
  component?: ElementType;
} & RcBaseProps<MuiTypographyProps, 'variant'>;
// https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/Typography/Typography.js
const MuiTypographyVariant: RcTypographyVariant[] = [
  'h1',
  'h2',
  'h3',
  'h4',
  'h5',
  'h6',
  'subtitle1',
  'subtitle2',
  'body1',
  'body2',
  'caption',
  'button',
  'overline',
  'srOnly',
  'inherit',
];
const WrappedMuiTypography = React.forwardRef(
  (
    {
      variant,
      component,
      ...rest
    }: Omit<RcTypographyProps, 'variant'> & { variant?: any },
    ref,
  ) => {
    return (
      <MuiTypography
        {...rest}
        component={component as ElementType}
        ref={ref}
        variant={
          variant && MuiTypographyVariant.includes(variant)
            ? variant
            : undefined
        }
      />
    );
  },
);

/** @deprecated please not use this component anymore */
const RcTypography = styled(WrappedMuiTypography)`
  && {
    ${({ variant }) => (variant === 'inherit' ? '' : typography(variant))}
  }
`;

RcTypography.displayName = 'RcTypography';
RcTypography.defaultProps = {
  variant: 'body1',
};

export { RcTypographyProps, RcTypography };
