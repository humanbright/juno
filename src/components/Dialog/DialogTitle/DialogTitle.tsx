import MuiDialogTitle from '@material-ui/core/DialogTitle';
import React, { ComponentProps, forwardRef, useMemo } from 'react';

import {
  combineClasses,
  RcBaseProps,
  styled,
  useThemeProps,
} from '../../../foundation';
import { RcTypography } from '../../Typography';
import { RcDialogChildrenProps, useDialogDefaultProps } from '../utils';
import { DialogTitleStyle } from './styles';
import { RcDialogTitleClasses, RcDialogTitleTypographyVariant } from './utils';

type RcDialogTitleProps = {
  display?: 'block' | 'flex';
  /** Define the padding size of that group wrapper. */
  space?: number | number[];
} & RcDialogChildrenProps &
  RcBaseProps<ComponentProps<typeof MuiDialogTitle>>;

const _RcDialogTitle = forwardRef<any, RcDialogTitleProps>(
  (inProps: RcDialogTitleProps, ref) => {
    const props = useThemeProps({ props: inProps, name: 'RcDialogTitle' });

    const {
      classes: classesProp,
      display,
      space,
      children,
      disableTypography,
      size,
      ...rest
    } = props;

    const classes = useMemo(
      () => combineClasses(RcDialogTitleClasses, classesProp),
      [classesProp],
    );

    return (
      <MuiDialogTitle {...rest} disableTypography ref={ref} classes={classes}>
        {disableTypography ? (
          children
        ) : (
          <RcTypography
            variant={RcDialogTitleTypographyVariant[size!]}
            component="h2"
          >
            {children}
          </RcTypography>
        )}
      </MuiDialogTitle>
    );
  },
);

const RcDialogTitle = styled(_RcDialogTitle).attrs(useDialogDefaultProps)`
  ${DialogTitleStyle}
`;

RcDialogTitle.defaultProps = {};

RcDialogTitle.displayName = 'RcDialogTitle';

export { RcDialogTitle, RcDialogTitleProps };
