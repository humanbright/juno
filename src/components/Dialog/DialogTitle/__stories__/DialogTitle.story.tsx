import { Meta, Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';

import Close from '../../../../icon/Close';
import Settings from '../../../../icon/Settings';
import Star from '../../../../icon/Star';
import {
  notControlInDocTable,
  notShowInDocTable,
  sortInDocTable,
  switchToControlKnobs,
} from '../../../../storybook';
import { RcAvatar } from '../../../Avatar';
import { RcBox } from '../../../Box';
import { RcIconButton } from '../../../Buttons/IconButton';
import { RcIconButtonGroup } from '../../../Buttons/IconButtonGroup';
import { RcTypography } from '../../../Typography';
import { RcDialogTitle } from '../DialogTitle';

export default {
  title: 'ðŸš€ Cleanup Components/Dialog/DialogTitle',
  component: RcDialogTitle,
  argTypes: {
    ...sortInDocTable<keyof DialogTitleProps>([]),
    ...notControlInDocTable<keyof DialogTitleProps>([]),
    ...notShowInDocTable<keyof DialogTitleProps>([]),
  },
} as Meta;

type DialogTitleProps = ComponentProps<typeof RcDialogTitle>;

export const DialogTitle: Story<DialogTitleProps> = ({ children, ...args }) => {
  switchToControlKnobs();
  return <RcDialogTitle {...args}>{children}</RcDialogTitle>;
};

DialogTitle.storyName = 'DialogTitle';

DialogTitle.args = {
  children: 'Title',
};

DialogTitle.argTypes = {
  ...notControlInDocTable<keyof DialogTitleProps>([]),
};

DialogTitle.parameters = {
  tags: [
    {
      name: 'Mui',
      href: 'https://material-ui.com/api/dialog-title/',
    },
    {
      name: 'Accessibility',
      value: '100%',
    },
  ],
};

export const DialogTitleExamples: Story<DialogTitleProps> = ({
  children,
  ...args
}) => {
  switchToControlKnobs();
  return (
    <RcDialogTitle display="flex" disableTypography {...args} space={[0, 6]}>
      <RcAvatar size="medium" color="avatar.lake">
        SH
      </RcAvatar>
      <RcBox
        display="flex"
        justifyContent="center"
        flexDirection="column"
        margin="0 16px 0 8px"
      >
        <RcTypography variant="body1" color="neutral.f06">
          Jessica Lewis
        </RcTypography>
        <RcTypography variant="caption1" color="neutral.f03">
          1/23/2019 10:02 AM
        </RcTypography>
      </RcBox>
      <RcBox flex="1 1 auto" alignItems="center" display="flex">
        <RcTypography variant="subheading2" component="h2">
          {children}
        </RcTypography>
      </RcBox>
      <RcIconButtonGroup gap={-2.5}>
        <RcIconButton color="warning.f01" title="Favorite" symbol={Star} />
        <RcIconButton title="Settings" symbol={Settings} />
        <RcIconButton title="Close" symbol={Close} />
      </RcIconButtonGroup>
    </RcDialogTitle>
  );
};

DialogTitleExamples.storyName = 'DialogTitle Examples';
