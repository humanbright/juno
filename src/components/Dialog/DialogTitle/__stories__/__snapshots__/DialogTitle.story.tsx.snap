// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Dialog/DialogTitle DialogTitle 1`] = `
.c1 {
  font-size: 1.25rem;
  font-weight: 700;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 28px;
}

.c0 {
  color: #121212;
  padding: 16px 24px;
}

<div
  className="MuiDialogTitle-root c0"
>
  <h2
    className="MuiTypography-root c1"
    data-color="initial"
    data-variant="title2"
  >
    Title
  </h2>
</div>
`;

exports[`Storyshots ðŸš€ Cleanup Components/Dialog/DialogTitle DialogTitle Examples 1`] = `
.c10 {
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
  pointer-events: none;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 24px;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 40px;
  height: 40px;
  color: #F6AD16;
  border-radius: 50%;
  -webkit-transition: background 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  cursor: pointer;
}

.c7,
.c7 .c8 {
  font-size: 20px;
}

.c7:active,
.c7:active .c8 {
  color: #F6AD16;
}

.c7.focus-visible,
.c7[data-focus-visible-added] {
  background-color: rgba(246,173,22,0.16);
}

.c7.focus-visible:active,
.c7[data-focus-visible-added]:active,
.c7.focus-visible:active .c8,
.c7[data-focus-visible-added]:active .c8 {
  color: rgba(246,173,22,0.88);
}

.c7.RcIconButton-persistBg {
  background-color: rgba(246,173,22,0.12);
}

.c7.RcIconButton-disabled,
.c7.RcIconButton-disabled .c8 {
  color: #B9B9B9;
}

.c7.RcIconButton-invisible {
  opacity: 0;
  visibility: hidden;
  width: 0;
  height: 0;
}

.c7.RcIconButton-outline {
  border: 1px solid #C7C7C7;
}

.c7.RcIconButton-contained {
  color: rgba(0,0,0,0.87);
  background-color: #F6AD16;
}

.c7.RcIconButton-contained.focus-visible,
.c7.RcIconButton-contained[data-focus-visible-added] {
  background-color: rgba(246,173,22,0.84);
}

.c7.RcIconButton-contained.focus-visible,
.c7.RcIconButton-contained[data-focus-visible-added],
.c7.RcIconButton-contained.focus-visible .c8,
.c7.RcIconButton-contained[data-focus-visible-added] .c8 {
  color: rgba(0,0,0,0.87);
}

.c7.RcIconButton-contained:active {
  background-color: rgba(246,173,22,0.76);
}

.c7.RcIconButton-contained:active,
.c7.RcIconButton-contained:active .c8 {
  color: rgba(0,0,0,0.87);
}

.c7 .RcIconButtonTouchRipple-ripplePulsate {
  border-radius: 0;
  -webkit-animation-name: none;
  animation-name: none;
  opacity: 0;
}

.c11 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 40px;
  height: 40px;
  color: #666666;
  border-radius: 50%;
  -webkit-transition: background 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: background 150ms cubic-bezier(0.4,0,0.2,1) 0ms;
  cursor: pointer;
}

.c11,
.c11 .c8 {
  font-size: 20px;
}

.c11:active,
.c11:active .c8 {
  color: #666666;
}

.c11.focus-visible,
.c11[data-focus-visible-added] {
  background-color: rgba(102,102,102,0.16);
}

.c11.focus-visible:active,
.c11[data-focus-visible-added]:active,
.c11.focus-visible:active .c8,
.c11[data-focus-visible-added]:active .c8 {
  color: rgba(102,102,102,0.88);
}

.c11.RcIconButton-persistBg {
  background-color: rgba(102,102,102,0.12);
}

.c11.RcIconButton-disabled,
.c11.RcIconButton-disabled .c8 {
  color: #B9B9B9;
}

.c11.RcIconButton-invisible {
  opacity: 0;
  visibility: hidden;
  width: 0;
  height: 0;
}

.c11.RcIconButton-outline {
  border: 1px solid #C7C7C7;
}

.c11.RcIconButton-contained {
  color: #fff;
  background-color: #666666;
}

.c11.RcIconButton-contained.focus-visible,
.c11.RcIconButton-contained[data-focus-visible-added] {
  background-color: rgba(102,102,102,0.84);
}

.c11.RcIconButton-contained.focus-visible,
.c11.RcIconButton-contained[data-focus-visible-added],
.c11.RcIconButton-contained.focus-visible .c8,
.c11.RcIconButton-contained[data-focus-visible-added] .c8 {
  color: #fff;
}

.c11.RcIconButton-contained:active {
  background-color: rgba(102,102,102,0.76);
}

.c11.RcIconButton-contained:active,
.c11.RcIconButton-contained:active .c8 {
  color: #fff;
}

.c11 .RcIconButtonTouchRipple-ripplePulsate {
  border-radius: 0;
  -webkit-animation-name: none;
  animation-name: none;
  opacity: 0;
}

.c2 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  outline: none;
  color: #FFFFFF;
  font-size: 1.25rem;
  font-weight: 700;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 28px;
  width: 64px;
  height: 64px;
  background-color: #1A70C1;
}

.c2 > img,
.c2 > canvas {
  width: 100%;
  height: 100%;
}

.c1 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  outline: 0;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  white-space: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 > * + * {
  margin-left: -10px;
}

.c3 {
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
}

.c4 {
  font-size: 0.75rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 16px;
  color: #757575;
}

.c5 {
  font-size: 1rem;
  font-weight: 700;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 24px;
}

.c0 {
  color: #121212;
  padding: 0px 24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (hover:hover) and (pointer:fine) {
  .c7:hover {
    background-color: rgba(246,173,22,0.08);
  }

  .c7:hover,
  .c7:hover .c8 {
    color: rgba(246,173,22,0.88);
  }
}

@media (hover:hover) and (pointer:fine) {
  .c7.RcIconButton-contained:hover {
    background-color: rgba(246,173,22,0.92);
  }

  .c7.RcIconButton-contained:hover,
  .c7.RcIconButton-contained:hover .c8 {
    color: rgba(0,0,0,0.87);
  }
}

@media (hover:hover) and (pointer:fine) {
  .c11:hover {
    background-color: rgba(102,102,102,0.08);
  }

  .c11:hover,
  .c11:hover .c8 {
    color: rgba(102,102,102,0.88);
  }
}

@media (hover:hover) and (pointer:fine) {
  .c11.RcIconButton-contained:hover {
    background-color: rgba(102,102,102,0.92);
  }

  .c11.RcIconButton-contained:hover,
  .c11.RcIconButton-contained:hover .c8 {
    color: #fff;
  }
}

<div
  className="MuiDialogTitle-root c0"
>
  <div
    className="c1 "
    data-test-automation-class="avatar"
  >
    <div
      className="c2 RcAvatar-avatarContainer"
    >
      SH
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-130"
  >
    <p
      className="MuiTypography-root c3"
      data-color="neutral.f06"
      data-variant="body1"
    >
      Jessica Lewis
    </p>
    <p
      className="MuiTypography-root c4"
      data-color="neutral.f03"
      data-variant="caption1"
    >
      1/23/2019 10:02 AM
    </p>
  </div>
  <div
    className="MuiBox-root MuiBox-root-131"
  >
    <h2
      className="MuiTypography-root c5"
      data-color="initial"
      data-variant="subheading2"
    />
  </div>
  <div
    className="c6"
    direction="horizontal"
    onClick={[Function]}
    onKeyUp={[Function]}
  >
    <button
      aria-describedby={null}
      aria-label="Favorite"
      className="MuiButtonBase-root c7 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Favorite"
      type="button"
    >
      <span
        className="c8 c9 star icon"
      >
        <svg
          className="c10"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.885 1.726c.394.195.714.514.908.908l2.97 6.018a1 1 0 00.753.547l6.641.965a2.001 2.001 0 011.109 3.412L24.46 18.26c-.236.23-.343.561-.288.885l1.134 6.614a1.999 1.999 0 01-2.902 2.108l-5.94-3.123a1.002 1.002 0 00-.931 0l-5.94 3.123a2 2 0 01-2.902-2.109l1.134-6.614a.999.999 0 00-.288-.885l-4.806-4.684a2 2 0 011.109-3.411l6.641-.965a1 1 0 00.753-.547l2.97-6.018a2 2 0 012.679-.908z"
          />
        </svg>
        
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-describedby={null}
      aria-label="Settings"
      className="MuiButtonBase-root c11 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Settings"
      type="button"
    >
      <span
        className="c8 c9 settings icon"
      >
        <svg
          className="c10"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.857 2.057a15.048 15.048 0 014.823 2.688c.357.296.477.786.297 1.21a4.13 4.13 0 00.241 3.731 4.29 4.29 0 003.411 2.095c.467.033.858.359.966.806a14.528 14.528 0 01.001 6.827c-.108.448-.5.774-.967.807a4.285 4.285 0 00-3.411 2.094 4.132 4.132 0 00-.241 3.732c.18.423.06.913-.296 1.209a15.044 15.044 0 01-4.822 2.689 1.081 1.081 0 01-1.249-.428c-.783-1.2-2.13-1.939-3.608-1.939s-2.825.739-3.61 1.939a1.08 1.08 0 01-1.248.427 15.014 15.014 0 01-4.822-2.687 1.04 1.04 0 01-.297-1.21 4.13 4.13 0 00-.241-3.731 4.29 4.29 0 00-3.411-2.095 1.066 1.066 0 01-.966-.806 14.528 14.528 0 01-.001-6.827c.108-.448.5-.774.967-.807a4.285 4.285 0 003.411-2.094 4.135 4.135 0 00.242-3.732 1.04 1.04 0 01.296-1.209 15.044 15.044 0 014.822-2.689c.467-.157.982.02 1.249.428a4.295 4.295 0 003.608 1.94c1.458 0 2.789-.719 3.562-1.867l.038-.06a1.08 1.08 0 011.257-.441zM16 12a4 4 0 100 8 4 4 0 000-8z"
          />
        </svg>
        
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-describedby={null}
      aria-label="Close"
      className="MuiButtonBase-root c11 RcIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Close"
      type="button"
    >
      <span
        className="c8 c9 close icon"
      >
        <svg
          className="c10"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M25.899 7.515L17.414 16l8.485 8.486a1 1 0 01-1.414 1.414L16 17.413 7.515 25.9a1 1 0 01-1.414-1.414l8.485-8.487-8.485-8.484a.999.999 0 111.414-1.414l8.486 8.485 8.485-8.485A.999.999 0 1125.9 7.515z"
          />
        </svg>
        
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
